pixeltable/catalog/path.py

```diff
@@ -9,14 +10,9 @@
 _logger = logging.getLogger('pixeltable')
 
 
-class Path:
+class Path(NamedTuple):
     components: list[str]
-    version: int | None
-
-    def __init__(self, components: list[str], version: int | None = None) -> None:
-        assert len(components) > 0
```

Comment:
Can we have this assertion back?

Reply:
Yes. In a `NamedTuple` the `__init__()` method is still called, but after the fields are populated, so @amithadke I believe the relevant code should look like this:
```
def __init__(self) -> None:
    assert len(self.components) > 0
```
As an aside, we should also change `components` to a `tuple[str, ...]` instead of a `list[str]` (it's more performant and also more correct, since `components` is immutable). I *believe* we could even make it a `tuple[str, str, ...]` and doing so would give a type guarantee that `len(components) > 0`. (That doesn't need to be done as part of this PR)